### PR TITLE
Update env, cmake, and MAPL components to latest versions

### PR DESCRIPTION
**Pull request by mathomp4**
This PR updates the GEOSldas to be "in sync" with the latest GCM. This includes:

* ESMA_env v3.2.1
* ESMA_cmake v3.4.2
* MAPL v2.6.7

I also cleaned up the main `CMakeLists.txt` as it had some anachronisms and was missing out on some neat functionality (automatically ignore build and install directories no matter the name).

I do not see any reason these should cause any differences. But of course, @biljanaorescanin should test to make sure.

I did a test build just to make sure that would work.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -43,3 +40,17 @@ include_directories(${MPI_Fortran_INCLUDE_PATH})
 # Recursively build source tree
 add_subdirectory (@env)
 add_subdirectory (src)
+
+# https://www.scivision.dev/cmake-auto-gitignore-build-dir/
+# --- auto-ignore build directory
+if(NOT EXISTS ${PROJECT_BINARY_DIR}/.gitignore)
```
Comment (tclune) [inline CMakeLists.txt:46]: It's a fixture, so this is probably safe, but `${GEOSldas_BINARY_DIR}` is generally safer.

**Review by tclune — DISMISSED** [review]

**Review by gmao-rreichle — COMMENTED** [review]
@mathomp4 , @weiyuan-jiang , @biljanaorescanin ,
Should we merge this branch with that of PR #[381](https://github.com/GEOS-ESM/GEOSldas/pull/381) before testing? 
I just updated the latter to MAPL 2.6.7.  Presumably, updating ESMA_env and ESMA_cmake will be 0-diff as well.  Then we only need to run the tests once.

**Comment by biljanaorescanin** [comment]
I really don't have a preference, but if you need it sooner than later, running tests once will be faster. If there is no hurry I can test each PR separately.

**Comment by gmao-rreichle** [comment]
I'll defer to @mathomp4  and @weiyuan-jiang on this.

**Comment by mathomp4** [comment]
@gmao-rreichle Well, I did see a bug in the `components.yaml` of @weiyuan-jiang 's branch. I suppose he could merge this branch into his or copy over the two files I changed. Then I guess it's equivalent as far as my changes.

**Comment by biljanaorescanin** [comment]
It it helps in decision,  I am running tests on this PR and 6/10 are done and all passed so far.

**Comment by gmao-rreichle** [comment]
> @gmao-rreichle Well, I did see a bug in the `components.yaml` of @weiyuan-jiang 's branch. I suppose he could merge this branch into his or copy over the two files I changed. Then I guess it's equivalent as far as my changes.

Yes, my suggestion was to merge Matt's branch (feature/mathomp4/update-ldas-components) into Weiyuan's branch (feature/wjiang/with_oserver).  Let's wait for Weiyuan to weigh in.  It's his branch after all. 
And yes, I was wondering what it might mean to have both a branch and a tag in components.yaml.  Glad to know I won't have to understand that one anymore. 
cc: @weiyuan-jiang @mathomp4 @biljanaorescanin

**Comment by weiyuan-jiang** [comment]
Considering the bug in MAPL v2.6.7 that mis-matched ESMF_Initialize() and ESMF_Finalize(), I will updated it to MAPL_v2.6.8

**Comment by mathomp4** [comment]
> Considering the bug in MAPL v2.6.7 that mis-matched ESMF_Initialize() and ESMF_Finalize(), I will updated it to MAPL_v2.6.8

Oh yeah, please do. I made this PR before we had v2.6.8.

**Comment by weiyuan-jiang** [comment]
I have merged this PR to PR #381 . So we can close this PR now

**Comment by gmao-rreichle** [comment]
Closing, will be addressed in #381.  See @weiyuan-jiang's message above.

**Pull request closed without merging**